src/bson/bson.c

```diff
@@ -2728,9 +2728,16 @@ _bson_as_json_visit_regex (const bson_iter_t *iter,
 {
    bson_json_state_t *state = data;
    const char *c;
+   char *escaped_regex;
+
+   escaped_regex = bson_utf8_escape_for_json (v_regex, -1);
```

Comment:
As with whitespace and key order, tests need to account for legal differences between the corpus text and the generated text.  For example, in Perl, I can "wash" the JSON texts through a decode->encode cycle that normalizes differences.  For C, I imagine that tests will need to iterate keys and compare values directly in some way that accounts for character escaping differences while still ensuring that all *required* characters are escaped.